go/interpreter/lfvm/gas.go

```diff
@@ -316,131 +310,71 @@ func callGas(availableGas, base tosca.Gas, callCost *uint256.Int) tosca.Gas {
 	return tosca.Gas(callCost.Uint64())
 }
 
-// Computes the costs for an SSTORE operation
-func gasSStore(c *context) (tosca.Gas, error) {
-	return gasSStoreEIP2200(c)
-}
+func getDynamicCostsForSstore(
+	revision tosca.Revision,
+	storageStatus tosca.StorageStatus,
+) tosca.Gas {
+	isBerlinOrNewer := revision >= tosca.R09_Berlin
 
-//  0. If *gasleft* is less than or equal to 2300, fail the current call.
-//  1. If current value equals new value (this is a no-op), SLOAD_GAS is deducted.
-//  2. If current value does not equal new value:
-//     2.1. If original value equals current value (this storage slot has not been changed by the current execution context):
-//     2.1.1. If original value is 0, SSTORE_SET_GAS (20K) gas is deducted.
-//     2.1.2. Otherwise, SSTORE_RESET_GAS gas is deducted. If new value is 0, add SSTORE_CLEARS_SCHEDULE to refund counter.
-//     2.2. If original value does not equal current value (this storage slot is dirty), SLOAD_GAS gas is deducted. Apply both of the following clauses:
-//     2.2.1. If original value is not 0:
-//     2.2.1.1. If current value is 0 (also means that new value is not 0), subtract SSTORE_CLEARS_SCHEDULE gas from refund counter.
-//     2.2.1.2. If new value is 0 (also means that current value is not 0), add SSTORE_CLEARS_SCHEDULE gas to refund counter.
-//     2.2.2. If original value equals new value (this storage slot is reset):
-//     2.2.2.1. If original value is 0, add SSTORE_SET_GAS - SLOAD_GAS to refund counter.
-//     2.2.2.2. Otherwise, add SSTORE_RESET_GAS - SLOAD_GAS gas to refund counter.
-func gasSStoreEIP2200(c *context) (tosca.Gas, error) {
-	// If we fail the minimum gas availability invariant, fail (0)
-	if c.gas <= SstoreSentryGasEIP2200 {
-		c.status = statusOutOfGas
-		return 0, errNotEnoughGasReentrancy
+	cost := tosca.Gas(800)
```

Comment:
this was quite confusing and it took me some time to realize that the switch in line 324 is not exhaustive. 
Nit: I would move this declaration and the if isberlinOrNewer if into a swtich default clause. to avoid mutation of state (cost varaible),  and having a more declarative function flow.

Reply:
Yes, good idea, applied to both functions.